akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -420,6 +420,7 @@ private final case class SavedIslandData(islandGlobalOffset: Int, lastVisitedOff
     defaultAttributes: Attributes,
     defaultPhase:      Phase[Any],
     phases:            Map[IslandTag, Phase[Any]]): Mat = {
+    if (isShutdown) throw new IllegalStateException("Trying to materialize stream after materializer has been shutdown")
```

Comment:
what would happen if the materialization starts in one thread, and half ways through another thread calls `shutdown`?

Reply:
It has sent a `PoisonPill` to the supervisor, so it would maybe not throw an IllegalState, but it shouldn't leak anything. Maybe we should check in the end as well to be able to provide this clear error rather than potentially fail silently

Reply:
sounds good to do the same check at the end of this method